### DISCUSSION TITLE
Seaport Usage

### DISCUSSION THREAD

**Discussion by pepperminting** [ProjectOpenSea/seaport#566] [Q&A] [answered]
1. Can I fork this code and edit it for my own NFT marketplace?

2. Am I allowed to create my own order that doesn't include a consideration for paying 2.5% service fee to the OpenSea wallet then use deployed Seaport contracts such as ConduitController, Seaport or Conduit for my own NFT marketplace?

**Comment by Slokh** [comment, marked as answer]
1. You don't need to fork the code. You can just use the contract that has already been deployed https://github.com/ProjectOpenSea/seaport#deployments

2. You can do that, but the order won't be able to be shown on OpenSea if there is no consideration for our fee.

**Reply by howardpen9** [reply]
can you deploy to BSC?

**Reply by 0xAskar** [reply]
also, how do you create an offer/listing on opensea via the sdk without including a recipient? I'm getting bigNumber errors with my code. I left recipient empty in the first consideration because I just want to list, not send it to a specific address quite yet.
```
const provider = new ethers.providers.Web3Provider(window.ethereum);
            const seaport = new Seaport(provider);
            let buyer = await getBuyer()
            // create offer parameters
            let parameters = {
                offerer: buyer,
                zone: "0x004c00500000ad104d7dbd00e3ae0a5c00560c00",
                zoneHash: "0x3000000000000000000000000000000000000000000000000000000000000000",
                startTime: "0",
                endTime: (Math.floor(new Date().getTime()) + 7*24*60*60*1000).toString(),
                orderType: 0,
                offer: [
                    {
                        "itemType": 2,
                        "token": contract_address,
                        "identifierOrCriteria": token_id.toString(),
                        "startAmount": "1",
                        "endAmount": "1"
                    }
                ],
                "consideration": [
                    {
                        "itemType": 0,
                        "token": "0x0000000000000000000000000000000000000000",
                        "identifierOrCriteria": "0",
                        "startAmount": "93500000000000000",
                        "endAmount": "93500000000000000",
                        "recipient": ""
                    },
                    {
                        "itemType": 0, 
                        "token": "0x0000000000000000000000000000000000000000",
                        "identifierOrCriteria": "0",
                        "startAmount": "250000000000000000",
                        "endAmount": "250000000000000000",
                        "recipient": "0x8de9c5a032463c561423387a9648c5c7bcc5bc90",
                    },
                    {
                        "itemType": 0, 
                        "token": "0x0000000000000000000000000000000000000000",
                        "identifierOrCriteria": "0",
                        "startAmount": "250000000000000000",
                        "endAmount": "250000000000000000",
                        "recipient": "0x4c1cd907ceaa5919cf7982679fce88c58e423dcb",
                    }

                ]
            }
            console.log(parameters)
            const {executeAllActions} = await seaport.createOrder(parameters)
            const executedOrder = await executeAllActions()
            console.log(executedOrder)
        }
```

**Reply by pamungkaski** [reply]
@0xAskar you need recipients, it is for when the listing is bought, ricipients will be the address in which the fee will be sent, so it would be the offerer address usually

**Reply by 0xAskar** [reply]
hey, I was wondering how would you be able to then create a listing _using_ the seaport documentation though _without_ fulfilling an order. any thoughts?

**Comment by pguardiario** [comment]
@slokh - can we have an example of what the consideration should look like to show up on opensea?
In other words, an example of listing an item that will show on opensea.
Thanks.

**Reply by njovy** [reply]
@pguardiario Please check [this link](https://docs.opensea.io/v2.0/reference/create-an-order-testnets ). You need to use this API to show your listing on Opensea. Otherwise it won't be visible. If you don't include 2.5% fee, the Opensea server won't proceed your request. Set a recipient of a consideration item to 0x8De9C5A032463C561423387a9648c5C7BCC5BC90 which is the address of Opensea: Fee. Then set 2.5% of total to startAmount and endAmount. If a collection has a creator earning, you will have to set that as well. Errors are pretty straightforward if your request is invalid so easy to fix it.

**Reply by 0xAskar** [reply]
See, I did this, and I'm still getting a bad request. I'm trying to list [this nft.](https://opensea.io/assets/ethereum/0x569f26145f0c2fcc3da68395be3639009c53b6d8/546) Here's the code:
```
let parameters = {
    conduitKey: "0x0000000000000000000000000000000000000000000000000000000000000000",
    consideration: [{
        endAmount: "97500000000000000",
        identifierOrCriteria: "0",
        itemType: 0,
        recipient: "0xDF93B5ec8Eb9407F03fF14a2E2B3eE4196fcD8b0",
        startAmount: "97500000000000000",
        token: "0x0000000000000000000000000000000000000000"
    },
    {   
        endAmount: "2500000000000000",
        identifierOrCriteria: "0",
        itemType: 0,
        recipient: "0x8De9C5A032463C561423387a9648c5C7BCC5BC90",
        startAmount: "2500000000000000",
        token: "0x0000000000000000000000000000000000000000",
    },
    {   
        endAmount: "4000000000000000",
        identifierOrCriteria: "0",
        itemType: 0,
        recipient: "0xbeb423ffe37c9b63314f5a3043b28d2d8631b4d9",
        startAmount: "4000000000000000",
        token: "0x0000000000000000000000000000000000000000",
    }
    ],
    counter: 0,
    endTime: "1658771617000",
    offer: [{
        endAmount: "1",
        identifierOrCriteria: "546",
        itemType: 2,
        startAmount: "1",
        token: "0x569f26145f0c2fcc3da68395be3639009c53b6d8",
    }],
    offerer: "0xDF93B5ec8Eb9407F03fF14a2E2B3eE4196fcD8b0",
    orderType: 0,
    salt: "0x91744fc98710dd16d957d2cbe8fb1cbe",
    startTime: "0",
    totalOriginalConsiderationItems: 3,
    zone: "0x0000000000000000000000000000000000000000",
    zoneHash: "0x3000000000000000000000000000000000000000000000000000000000000000",
    // nonce: "725"
}
```

**Comment by pguardiario** [comment]
Thanks for the response @njovy.
I'm actually interested in doing this with the seaport library (not the api).
@slokh seems to indicate that it is possible.

**Reply by 0xAskar** [reply]
yes, does anyone know how to use the seaport documentation to create a listing, would you send the `order` without a buyer in the fulfillOrder?

**Reply by MiguelGonzales-WIV** [reply]
I have the same thoughts with @0xAskar. I'd need to indicate the fulfiller/buyer in fulfillOrder to create a listing, but I don't really see the point if we're just listing it? I don't know what to put there

**Reply by 0xAskar** [reply]
Yeah, I tried to leave it out, but it requires me to still sign a transaction and pay fulFillment fees. So, I figured it wasn't correct. Also, my friend pointed out that you may not be able to use Seaport because they just have a database of transactions and thus, the api would be the only method. However, as @Slokh pointed out, it seems feasible with Seaport
